src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java

```diff
@@ -265,80 +326,205 @@ private void setGracefulShutdownTimeout(long gracefulShutdownTimeout) {
     this.gracefulShutdownTimeout = gracefulShutdownTimeout;
   }
 
+  @Override
+  public SchedulerPoolStrategy getSchedulerPoolStrategy() {
+    return schedulerPoolStrategy;
+  }
+
+  public void setSchedulerPoolStrategy(SchedulerPoolStrategy schedulerPoolStrategy) {
+    setSchedulerPoolStrategy(schedulerPoolStrategy, false);
+  }
+
+  public void setSchedulerPoolStrategy(SchedulerPoolStrategy schedulerPoolStrategy, boolean initDefaults) {
+    this.schedulerPoolStrategy = schedulerPoolStrategy;
+    if (initDefaults) {
+      if (schedulerPoolStrategy == DEDICATED) {
+        cpuLightQueueSize = 0;
+        cpuLightPoolSize = 2 * CORES;
+        ioQueueSize = 0;
+        ioCorePoolSize = CORES;
+        ioMaxPoolSize = (int) max(2, CORES + ((MEM - 245760) / 5120));
```

Comment:
this formula is repeated 3 times in this class. Refactor to have this and all other formulas only once.